File: test_v1beta.py
import os
import json
import urllib.request

API_KEY = os.environ.get('GEMINI_API_KEY')
url = f"https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key={API_KEY}"
data = {
    "contents": [{"parts": [{"text": "Write a tiny 2 key JSON object."}]}],
    "generationConfig": {
        "responseMimeType": "application/json"
    }
}
req = urllib.request.Request(url, data=json.dumps(data).encode('utf-8'), headers={'Content-Type': 'application/json'})
try:
    with urllib.request.urlopen(req) as response:
        print(response.read().decode('utf-8'))
except Exception as e:
    print("Error:", e)
    if hasattr(e, 'read'):
        print(e.read().decode('utf-8'))

File: test_v1beta.js
require('dotenv').config({ path: 'blog_publisher/.env' });
const GEMINI_API_KEY = process.env.GEMINI_API_KEY || 'AIzaSy...'; // Can't easily use .env unless I know it, but user has it stored in Supabase secrets.

async function test() {
    const reqBody = {
        contents: [{ role: "user", parts: [{ text: "Write {\"status\": \"ok\"}" }] }],
        generationConfig: {
            temperature: 0.8,
            responseMimeType: "application/json"
        }
    };

    const geminiRes = await fetch(
        `https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent?key=${process.env.TEST_API_KEY}`,
        { method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(reqBody) }
    );
    const text = await geminiRes.text();
    console.log(text);
}
test();
